3-extra/2-array-of-objects.js

```diff
@@ -9,74 +9,104 @@
     Implement a function which takes the array of books as a parameter, and returns an array of book titles.
     Each title in the resulting array should be the highest rated book in its genre.
 */
+let highestBooks = [];
+let highBook = {};
+let finalList = [];
 
-function getHighestRatedInEachGenre(books) {
-    // TODO
+function checkGenreListing(genreToCheck) {
+  for (let i = 0; i < highestBooks.length; i++) {
+    if (highestBooks[i].genre === genreToCheck) {
+      return i;
+    }
+  }
+  return -1;
 }
 
+function getHighestRatedInEachGenre(books) {
```

Comment:
Suggestion: After next class have a think about how this solution might be simplified by using an object.